dags/inference/maxtext_inference_microbenchmark.py

```diff
@@ -0,0 +1,313 @@
+          test_name=test_name,
+          test_mode=SetupMode.STABLE,
+          project_name=project_name,
+          runtime_version=runtime_version,
+          network=network,
+          subnetwork=subnetwork,
+          is_tpu_reserved=True,
+          model_configs=model_configs,
+          maxtext_branch=model_configs["maxtext_branch"],
+      )
+  )
+
+  return maxtext_kv_cache_layout_optimization
+
+
+dag_id = (
+    "maxtext-inference-microbenchmark"
+    if not USER_PREFIX
+    else f"{USER_PREFIX}-maxtext-inference-microbenchmark"
+)
+tags = ["inference_team", "maxtext", "microbenchmark"]
+
+if USER_PREFIX:
+  dag_id = f"{USER_PREFIX}-maxtext-inference-microbenchmark"
+  tags.append(USER_PREFIX)
+
+with models.DAG(
+    dag_id=dag_id,
+    tags=tags,
+    start_date=datetime.datetime(2024, 1, 19),
+    schedule=None,
+    catchup=False,
+) as dag:
+  test_name_prefix = (
+      "max-micro" if not USER_PREFIX else f"{USER_PREFIX}-max-micro"
+  )
+
+  sweep_vm_count = 8
+  (
+      two_axis_order_product_id_concat_list,
+      prefill_cache_axis_order_concat_list,
+      ar_cache_axis_order_concat_list,
+  ) = get_concatenated_list_of_params(sweep_vm_count=sweep_vm_count)
+
+  test_templates = {
+      LLAMA2_7B: {
+          "maxtext_branch": ""
+          if not MAXTEXT_BRANCH
+          else f"-b {MAXTEXT_BRANCH}",
+          "sleep_time": 60,
+          "tpu_version_cores": [(TpuVersion.V5E, 8)],
+          "model_name": LLAMA2_7B,
+          "tokenizer": "tokenizer.llama2",
+          "weight_dtype": "bfloat16",
+          "scan_layers": "false",
+          "max_prefill_predict_length": 1024,
+          "max_target_length": 2048,
+          "attention": "dot_product",
+          # (ici_fsdp_parallelism, ici_autoregressive_parallelism, ici_tensor_parallelism)
+          "ici_parallelisms": [(1, 1, -1)],
+          "inference_microbenchmark_prefill_lengths": "64,128,256,512,1024",
+          "inference_microbenchmark_stages": "prefill, generate",
+          "inference_microbenchmark_loop_iters": 10,
+          "base_output_directory": f"{BASE_OUTPUT_DIRECTORY}/{test_name_prefix}/kv_cache_layout_optimization/{test_run_datetime}",
+          "profiler": "xplane",
+          "save_config_to_gcs": "true",
+          "reshape_q": "true",
+          "compute_axis_order": ["0,2,1,3"],
+      },
+  }
+
+  tests = {
+      f"{LLAMA2_7B}-{W_BF16_KV_BF16}": test_templates[LLAMA2_7B]
+      | {
+          "quant_mode": W_BF16_KV_BF16,
+          "quantization": "",
+          "quantize_kvcache": "false",
+          "per_device_batch_size": 10,
+          "kv_quant_axis": "",
+          "time_out_in_min": 330,
+      },
+      f"{LLAMA2_7B}-{W_INT8_KV_INT8}": test_templates[LLAMA2_7B]
+      | {
+          "quant_mode": W_INT8_KV_INT8,
+          "quantization": "int8",
+          "quantize_kvcache": "true",
+          "per_device_batch_size": 24,
+          "kv_quant_axis": "kd",
```

Comment:
@morgandu I think the options will change from https://github.com/google/maxtext/pull/708. Should we use default or `dkv` now?

Reply:
We should change to heads_and_dkv

Reply:
Done. Thanks!